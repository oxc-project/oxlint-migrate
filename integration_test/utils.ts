import { expect } from 'vitest';
import main from '../src/index.js';
import { OxlintConfig } from '../src/types.js';

export const getSnapshotResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig?: OxlintConfig
) => {
  const collector: string[] = [];
  const result = await main(config, oxlintConfig, {
    reporter: collector.push.bind(collector),
    upgrade: oxlintConfig !== undefined,
  });

  return {
    config: result,
    warnings: collector
      // filter out unsupported rules
      .filter((error) => !error.startsWith('unsupported rule: local/'))
      .filter((error) => !error.startsWith('unsupported rule: perfectionist/'))
      .filter((error) => !error.startsWith('unsupported rule: vue/'))
      .filter((error) => !error.startsWith('unsupported rule: regexp/'))
      .filter((error) => !error.startsWith('unsupported rule: toml/'))
      .filter((error) => !error.startsWith('unsupported rule: style/')),
  };
};

export const getSnapShotUpgradeResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig: OxlintConfig
) => {
  const result = await getSnapshotResult(config, oxlintConfig);
  const upgradedResult = structuredClone(result);
  const result2 = await getSnapshotResult(config, upgradedResult.config);

  expect(result2).toStrictEqual(result);

  return result2;
};
