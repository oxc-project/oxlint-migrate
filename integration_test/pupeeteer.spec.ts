import { expect, test } from 'vitest';
// @ts-ignore
import puppeteer_test from './projects/puppeteer.eslint.config.mjs';
import { getSnapshotResult, getSnapShotUpgradeResult } from './utils.js';

test('puppeteer', async () => {
  const result = await getSnapshotResult(puppeteer_test);
  expect(result).toMatchSnapshot('puppeteer');
});

test('puppeteer upgrade', async () => {
  const result = await getSnapShotUpgradeResult(puppeteer_test, {
    categories: {
      correctness: 'error',
      perf: 'error',
    },
  });
  expect(result).toMatchSnapshot('puppeteer--upgrade');
});
