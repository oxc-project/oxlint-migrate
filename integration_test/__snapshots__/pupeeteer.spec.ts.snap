// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`puppeteer > puppeteer 1`] = `
{
  "config": {
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "categories": {
      "correctness": "off",
    },
    "env": {
      "builtin": true,
      "commonjs": true,
      "es2025": true,
      "node": true,
      "shared-node-browser": true,
    },
    "ignorePatterns": [
      "**/node_modules",
      "**/build/",
      "**/lib/",
      "**/bin/",
      "**/*.tsbuildinfo",
      "**/*.api.json",
      "**/*.tgz",
      "**/yarn.lock",
      "**/.docusaurus/",
      "**/.cache-loader",
      "test/output-*/",
      "**/.dev_profile*",
      "**/coverage/",
      "**/generated/",
      "**/.eslintcache",
      "**/.cache/",
      "**/.vscode",
      "!.vscode/extensions.json",
      "!.vscode/*.template.json",
      "**/.devcontainer",
      "**/.DS_Store",
      "**/.env.local",
      "**/.env.development.local",
      "**/.env.test.local",
      "**/.env.production.local",
      "**/npm-debug.log*",
      "**/yarn-debug.log*",
      "**/yarn-error.log*",
      "**/.wireit",
      "**/assets/",
      "**/third_party/",
      "packages/ng-schematics/sandbox/**/*",
      "packages/ng-schematics/multi/**/*",
      "packages/ng-schematics/src/**/files/",
      "examples/puppeteer-in-browser/out/**/*",
      "examples/puppeteer-in-browser/node_modules/**/*",
      "examples/puppeteer-in-extension/out/**/*",
      "examples/puppeteer-in-extension/node_modules/**/*",
    ],
    "overrides": [
      {
        "files": [
          "**/*.ts",
        ],
        "rules": {
          "no-class-assign": "off",
          "no-const-assign": "off",
          "no-dupe-class-members": "off",
          "no-dupe-keys": "off",
          "no-func-assign": "off",
          "no-import-assign": "off",
          "no-new-native-nonconstructor": "off",
          "no-obj-calls": "off",
          "no-redeclare": "off",
          "no-setter-return": "off",
          "no-this-before-super": "off",
          "no-unsafe-negation": "off",
          "no-with": "off",
          "prefer-rest-params": "error",
          "prefer-spread": "error",
        },
      },
      {
        "files": [
          "**/*.ts",
        ],
        "rules": {
          "no-class-assign": "off",
          "no-const-assign": "off",
          "no-dupe-class-members": "off",
          "no-dupe-keys": "off",
          "no-func-assign": "off",
          "no-import-assign": "off",
          "no-new-native-nonconstructor": "off",
          "no-obj-calls": "off",
          "no-redeclare": "off",
          "no-setter-return": "off",
          "no-this-before-super": "off",
          "no-unsafe-negation": "off",
          "no-with": "off",
          "prefer-rest-params": "error",
          "prefer-spread": "error",
        },
      },
      {
        "files": [
          "**/*.ts",
        ],
        "plugins": [
          "typescript",
        ],
        "rules": {
          "@typescript-eslint/ban-ts-comment": "error",
          "@typescript-eslint/no-duplicate-enum-values": "error",
          "@typescript-eslint/no-empty-object-type": "error",
          "@typescript-eslint/no-explicit-any": "error",
          "@typescript-eslint/no-extra-non-null-assertion": "error",
          "@typescript-eslint/no-misused-new": "error",
          "@typescript-eslint/no-namespace": "error",
          "@typescript-eslint/no-non-null-asserted-optional-chain": "error",
          "@typescript-eslint/no-require-imports": "error",
          "@typescript-eslint/no-this-alias": "error",
          "@typescript-eslint/no-unnecessary-type-constraint": "error",
          "@typescript-eslint/no-unsafe-declaration-merging": "error",
          "@typescript-eslint/no-unsafe-function-type": "error",
          "@typescript-eslint/no-wrapper-object-types": "error",
          "@typescript-eslint/prefer-as-const": "error",
          "@typescript-eslint/prefer-namespace-keyword": "error",
          "@typescript-eslint/triple-slash-reference": "error",
          "no-array-constructor": "error",
          "no-unused-expressions": "error",
          "no-unused-vars": "error",
        },
      },
      {
        "files": [
          "**/*.ts",
        ],
        "rules": {
          "no-class-assign": "off",
          "no-const-assign": "off",
          "no-dupe-class-members": "off",
          "no-dupe-keys": "off",
          "no-func-assign": "off",
          "no-import-assign": "off",
          "no-new-native-nonconstructor": "off",
          "no-obj-calls": "off",
          "no-redeclare": "off",
          "no-setter-return": "off",
          "no-this-before-super": "off",
          "no-unsafe-negation": "off",
          "no-with": "off",
          "prefer-rest-params": "error",
          "prefer-spread": "error",
        },
      },
      {
        "files": [
          "**/*.ts",
        ],
        "plugins": [
          "typescript",
        ],
        "rules": {
          "@typescript-eslint/adjacent-overload-signatures": "error",
          "@typescript-eslint/array-type": "error",
          "@typescript-eslint/ban-tslint-comment": "error",
          "@typescript-eslint/consistent-generic-constructors": "error",
          "@typescript-eslint/consistent-indexed-object-style": "error",
          "@typescript-eslint/consistent-type-definitions": "error",
          "@typescript-eslint/no-confusing-non-null-assertion": "error",
          "@typescript-eslint/no-inferrable-types": "error",
          "@typescript-eslint/prefer-for-of": "error",
          "@typescript-eslint/prefer-function-type": "error",
          "no-empty-function": "error",
        },
      },
      {
        "files": [
          "**/*.ts",
        ],
        "plugins": [
          "typescript",
        ],
        "rules": {
          "@typescript-eslint/array-type": [
            "error",
            {
              "default": "array-simple",
            },
          ],
          "@typescript-eslint/consistent-type-definitions": [
            "error",
            "interface",
          ],
          "@typescript-eslint/consistent-type-imports": "error",
          "@typescript-eslint/explicit-function-return-type": "off",
          "@typescript-eslint/no-explicit-any": "off",
          "@typescript-eslint/no-import-type-side-effects": "error",
          "@typescript-eslint/no-non-null-assertion": "off",
          "@typescript-eslint/no-unsafe-function-type": "error",
          "@typescript-eslint/no-wrapper-object-types": "error",
          "@typescript-eslint/prefer-ts-expect-error": "error",
          "no-empty-function": "off",
          "no-unused-vars": [
            "error",
            {
              "argsIgnorePattern": "^_",
              "varsIgnorePattern": "^_",
            },
          ],
        },
      },
      {
        "files": [
          "packages/puppeteer-core/src/**/*.ts",
        ],
        "rules": {
          "no-restricted-imports": [
            "error",
            {
              "paths": [],
              "patterns": [
                "*Events",
                "*.test.js",
              ],
            },
          ],
        },
      },
      {
        "files": [
          "test/**/*.ts",
        ],
        "rules": {
          "no-restricted-imports": [
            "error",
            {
              "patterns": [
                "*src*",
              ],
            },
          ],
        },
      },
      {
        "files": [
          "test/**/*.spec.ts",
        ],
        "plugins": [
          "typescript",
        ],
        "rules": {
          "no-unused-vars": [
            "error",
            {
              "argsIgnorePattern": "^_",
              "varsIgnorePattern": "^_",
            },
          ],
        },
      },
    ],
    "plugins": [
      "import",
    ],
    "rules": {
      "arrow-body-style": [
        "error",
        "always",
      ],
      "curly": [
        "error",
        "all",
      ],
      "eqeqeq": [
        "error",
      ],
      "import/no-cycle": [
        "error",
        {
          "maxDepth": Infinity,
        },
      ],
      "no-caller": "error",
      "no-cond-assign": "error",
      "no-debugger": "error",
      "no-dupe-keys": "error",
      "no-duplicate-case": "error",
      "no-empty-character-class": "error",
      "no-labels": "error",
      "no-multi-str": "error",
      "no-self-compare": "error",
      "no-shadow-restricted-names": "error",
      "no-unsafe-negation": "error",
      "no-unused-vars": [
        "error",
        {
          "args": "none",
          "vars": "local",
          "varsIgnorePattern": "([fx]?describe|[fx]?it|beforeAll|beforeEach|afterAll|afterEach)",
        },
      ],
      "no-var": "error",
      "no-with": "error",
      "radix": "error",
      "require-yield": "error",
      "valid-typeof": "error",
    },
  },
  "warnings": [
    "ignore allow list is currently not supported: !.vscode/extensions.json",
    "ignore allow list is currently not supported: !.vscode/*.template.json",
    "unsupported rule, but in development: import/named",
    "unsupported rule: prettier/prettier",
    "unsupported rule: spaced-comment",
    "unsupported rule: accessor-pairs",
    "unsupported rule: new-parens",
    "unsupported rule: prefer-const",
    "unsupported rule: max-len",
    "unsupported rule: no-implied-eval",
    "unsupported rule: no-new-object",
    "unsupported rule: no-octal-escape",
    "unsupported rule, but in development: no-unreachable",
    "unsupported rule: template-curly-spacing",
    "unsupported rule: mocha/no-exclusive-tests",
    "unsupported rule: import/order",
    "unsupported rule: @stylistic/func-call-spacing",
    "unsupported rule: @stylistic/semi",
    "unsupported rule: rulesdir/prettier-comments",
    "unsupported rule: rulesdir/extensions",
    "unsupported rule: rulesdir/check-license",
    "unsupported rule, but in development: getter-return",
    "unsupported rule, but in development: no-undef",
    "unsupported rule, but in development: no-unreachable",
    "unsupported rule: prefer-const",
    "unsupported rule, but in development: getter-return",
    "unsupported rule, but in development: no-undef",
    "unsupported rule, but in development: no-unreachable",
    "unsupported rule: prefer-const",
    "unsupported rule, but in development: getter-return",
    "unsupported rule, but in development: no-undef",
    "unsupported rule, but in development: no-unreachable",
    "unsupported rule: prefer-const",
    "unsupported rule: @typescript-eslint/class-literal-property-style",
    "unsupported rule: @typescript-eslint/consistent-type-assertions",
    "unsupported rule: rulesdir/use-using",
    "unsupported rule: @typescript-eslint/no-unnecessary-template-expression",
    "unsupported rule: @typescript-eslint/explicit-module-boundary-types",
    "unsupported rule: no-restricted-syntax",
    "unsupported rule: @typescript-eslint/no-floating-promises",
    "unsupported rule: @typescript-eslint/return-await",
    "unsupported rule: @typescript-eslint/consistent-type-exports",
    "unsupported rule: no-restricted-syntax",
    "unsupported rule: tsdoc/syntax",
    "unsupported rule: no-restricted-syntax",
    "unsupported rule: mocha/no-skipped-tests",
    "unsupported rule: mocha/no-identical-title",
  ],
}
`;

exports[`puppeteer merge > puppeteer--merge 1`] = `
{
  "config": {
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "categories": {
      "correctness": "error",
      "perf": "error",
    },
    "env": {
      "builtin": true,
      "commonjs": true,
      "es2025": true,
      "node": true,
      "shared-node-browser": true,
    },
    "ignorePatterns": [
      "**/node_modules",
      "**/build/",
      "**/lib/",
      "**/bin/",
      "**/*.tsbuildinfo",
      "**/*.api.json",
      "**/*.tgz",
      "**/yarn.lock",
      "**/.docusaurus/",
      "**/.cache-loader",
      "test/output-*/",
      "**/.dev_profile*",
      "**/coverage/",
      "**/generated/",
      "**/.eslintcache",
      "**/.cache/",
      "**/.vscode",
      "!.vscode/extensions.json",
      "!.vscode/*.template.json",
      "**/.devcontainer",
      "**/.DS_Store",
      "**/.env.local",
      "**/.env.development.local",
      "**/.env.test.local",
      "**/.env.production.local",
      "**/npm-debug.log*",
      "**/yarn-debug.log*",
      "**/yarn-error.log*",
      "**/.wireit",
      "**/assets/",
      "**/third_party/",
      "packages/ng-schematics/sandbox/**/*",
      "packages/ng-schematics/multi/**/*",
      "packages/ng-schematics/src/**/files/",
      "examples/puppeteer-in-browser/out/**/*",
      "examples/puppeteer-in-browser/node_modules/**/*",
      "examples/puppeteer-in-extension/out/**/*",
      "examples/puppeteer-in-extension/node_modules/**/*",
    ],
    "overrides": [
      {
        "files": [
          "**/*.ts",
        ],
        "rules": {
          "no-class-assign": "off",
          "no-const-assign": "off",
          "no-dupe-class-members": "off",
          "no-dupe-keys": "off",
          "no-func-assign": "off",
          "no-import-assign": "off",
          "no-new-native-nonconstructor": "off",
          "no-obj-calls": "off",
          "no-redeclare": "off",
          "no-setter-return": "off",
          "no-this-before-super": "off",
          "no-unsafe-negation": "off",
          "no-with": "off",
          "prefer-rest-params": "error",
          "prefer-spread": "error",
        },
      },
      {
        "files": [
          "**/*.ts",
        ],
        "rules": {
          "no-class-assign": "off",
          "no-const-assign": "off",
          "no-dupe-class-members": "off",
          "no-dupe-keys": "off",
          "no-func-assign": "off",
          "no-import-assign": "off",
          "no-new-native-nonconstructor": "off",
          "no-obj-calls": "off",
          "no-redeclare": "off",
          "no-setter-return": "off",
          "no-this-before-super": "off",
          "no-unsafe-negation": "off",
          "no-with": "off",
          "prefer-rest-params": "error",
          "prefer-spread": "error",
        },
      },
      {
        "files": [
          "**/*.ts",
        ],
        "plugins": [
          "typescript",
        ],
        "rules": {
          "@typescript-eslint/ban-ts-comment": "error",
          "@typescript-eslint/no-duplicate-enum-values": "error",
          "@typescript-eslint/no-empty-object-type": "error",
          "@typescript-eslint/no-explicit-any": "error",
          "@typescript-eslint/no-extra-non-null-assertion": "error",
          "@typescript-eslint/no-misused-new": "error",
          "@typescript-eslint/no-namespace": "error",
          "@typescript-eslint/no-non-null-asserted-optional-chain": "error",
          "@typescript-eslint/no-require-imports": "error",
          "@typescript-eslint/no-this-alias": "error",
          "@typescript-eslint/no-unnecessary-type-constraint": "error",
          "@typescript-eslint/no-unsafe-declaration-merging": "error",
          "@typescript-eslint/no-unsafe-function-type": "error",
          "@typescript-eslint/no-wrapper-object-types": "error",
          "@typescript-eslint/prefer-as-const": "error",
          "@typescript-eslint/prefer-namespace-keyword": "error",
          "@typescript-eslint/triple-slash-reference": "error",
          "no-array-constructor": "error",
          "no-unused-expressions": "error",
          "no-unused-vars": "error",
        },
      },
      {
        "files": [
          "**/*.ts",
        ],
        "rules": {
          "no-class-assign": "off",
          "no-const-assign": "off",
          "no-dupe-class-members": "off",
          "no-dupe-keys": "off",
          "no-func-assign": "off",
          "no-import-assign": "off",
          "no-new-native-nonconstructor": "off",
          "no-obj-calls": "off",
          "no-redeclare": "off",
          "no-setter-return": "off",
          "no-this-before-super": "off",
          "no-unsafe-negation": "off",
          "no-with": "off",
          "prefer-rest-params": "error",
          "prefer-spread": "error",
        },
      },
      {
        "files": [
          "**/*.ts",
        ],
        "plugins": [
          "typescript",
        ],
        "rules": {
          "@typescript-eslint/adjacent-overload-signatures": "error",
          "@typescript-eslint/array-type": "error",
          "@typescript-eslint/ban-tslint-comment": "error",
          "@typescript-eslint/consistent-generic-constructors": "error",
          "@typescript-eslint/consistent-indexed-object-style": "error",
          "@typescript-eslint/consistent-type-definitions": "error",
          "@typescript-eslint/no-confusing-non-null-assertion": "error",
          "@typescript-eslint/no-inferrable-types": "error",
          "@typescript-eslint/prefer-for-of": "error",
          "@typescript-eslint/prefer-function-type": "error",
          "no-empty-function": "error",
        },
      },
      {
        "files": [
          "**/*.ts",
        ],
        "plugins": [
          "typescript",
        ],
        "rules": {
          "@typescript-eslint/array-type": [
            "error",
            {
              "default": "array-simple",
            },
          ],
          "@typescript-eslint/consistent-type-definitions": [
            "error",
            "interface",
          ],
          "@typescript-eslint/consistent-type-imports": "error",
          "@typescript-eslint/explicit-function-return-type": "off",
          "@typescript-eslint/no-explicit-any": "off",
          "@typescript-eslint/no-import-type-side-effects": "error",
          "@typescript-eslint/no-non-null-assertion": "off",
          "@typescript-eslint/no-unsafe-function-type": "error",
          "@typescript-eslint/no-wrapper-object-types": "error",
          "@typescript-eslint/prefer-ts-expect-error": "error",
          "no-empty-function": "off",
          "no-unused-vars": [
            "error",
            {
              "argsIgnorePattern": "^_",
              "varsIgnorePattern": "^_",
            },
          ],
        },
      },
      {
        "files": [
          "packages/puppeteer-core/src/**/*.ts",
        ],
        "rules": {
          "no-restricted-imports": [
            "error",
            {
              "paths": [],
              "patterns": [
                "*Events",
                "*.test.js",
              ],
            },
          ],
        },
      },
      {
        "files": [
          "test/**/*.ts",
        ],
        "rules": {
          "no-restricted-imports": [
            "error",
            {
              "patterns": [
                "*src*",
              ],
            },
          ],
        },
      },
      {
        "files": [
          "test/**/*.spec.ts",
        ],
        "plugins": [
          "typescript",
        ],
        "rules": {
          "no-unused-vars": [
            "error",
            {
              "argsIgnorePattern": "^_",
              "varsIgnorePattern": "^_",
            },
          ],
        },
      },
    ],
    "plugins": [
      "import",
    ],
    "rules": {
      "arrow-body-style": [
        "error",
        "always",
      ],
      "curly": [
        "error",
        "all",
      ],
      "eqeqeq": [
        "error",
      ],
      "import/no-cycle": [
        "error",
        {
          "maxDepth": Infinity,
        },
      ],
      "no-labels": "error",
      "no-multi-str": "error",
      "no-self-compare": "error",
      "no-unused-vars": [
        "error",
        {
          "args": "none",
          "vars": "local",
          "varsIgnorePattern": "([fx]?describe|[fx]?it|beforeAll|beforeEach|afterAll|afterEach)",
        },
      ],
      "no-var": "error",
      "radix": "error",
    },
  },
  "warnings": [
    "ignore allow list is currently not supported: !.vscode/extensions.json",
    "ignore allow list is currently not supported: !.vscode/*.template.json",
    "unsupported rule, but in development: import/named",
    "unsupported rule: prettier/prettier",
    "unsupported rule: spaced-comment",
    "unsupported rule: accessor-pairs",
    "unsupported rule: new-parens",
    "unsupported rule: prefer-const",
    "unsupported rule: max-len",
    "unsupported rule: no-implied-eval",
    "unsupported rule: no-new-object",
    "unsupported rule: no-octal-escape",
    "unsupported rule, but in development: no-unreachable",
    "unsupported rule: template-curly-spacing",
    "unsupported rule: mocha/no-exclusive-tests",
    "unsupported rule: import/order",
    "unsupported rule: @stylistic/func-call-spacing",
    "unsupported rule: @stylistic/semi",
    "unsupported rule: rulesdir/prettier-comments",
    "unsupported rule: rulesdir/extensions",
    "unsupported rule: rulesdir/check-license",
    "unsupported rule, but in development: getter-return",
    "unsupported rule, but in development: no-undef",
    "unsupported rule, but in development: no-unreachable",
    "unsupported rule: prefer-const",
    "unsupported rule, but in development: getter-return",
    "unsupported rule, but in development: no-undef",
    "unsupported rule, but in development: no-unreachable",
    "unsupported rule: prefer-const",
    "unsupported rule, but in development: getter-return",
    "unsupported rule, but in development: no-undef",
    "unsupported rule, but in development: no-unreachable",
    "unsupported rule: prefer-const",
    "unsupported rule: @typescript-eslint/class-literal-property-style",
    "unsupported rule: @typescript-eslint/consistent-type-assertions",
    "unsupported rule: rulesdir/use-using",
    "unsupported rule: @typescript-eslint/no-unnecessary-template-expression",
    "unsupported rule: @typescript-eslint/explicit-module-boundary-types",
    "unsupported rule: no-restricted-syntax",
    "unsupported rule: @typescript-eslint/no-floating-promises",
    "unsupported rule: @typescript-eslint/return-await",
    "unsupported rule: @typescript-eslint/consistent-type-exports",
    "unsupported rule: no-restricted-syntax",
    "unsupported rule: tsdoc/syntax",
    "unsupported rule: no-restricted-syntax",
    "unsupported rule: mocha/no-skipped-tests",
    "unsupported rule: mocha/no-identical-title",
  ],
}
`;
